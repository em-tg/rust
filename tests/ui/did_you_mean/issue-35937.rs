//@ check-pass
struct Foo {
    pub v: Vec<String>
}

fn main() {
    let f = Foo { v: Vec::new() };
    f.v.push("cat".to_string()); //~ WARNING cannot borrow
}


struct S {
    x: i32,
}
fn foo() {
    let s = S { x: 42 };
    s.x += 1; //~ WARNING cannot assign
}

fn bar(s: S) {
    s.x += 1; //~ WARNING cannot assign
}
