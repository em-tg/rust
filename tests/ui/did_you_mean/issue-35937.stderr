warning[E0596]: cannot borrow `f.v` as mutable, as `f` is not declared as mutable
  --> $DIR/issue-35937.rs:8:5
   |
LL |     f.v.push("cat".to_string());
   |     ^^^ cannot borrow as mutable
   |
   = note: `#[warn(mut_non_mut)]` on by default
help: consider changing this to be mutable
   |
LL |     let mut f = Foo { v: Vec::new() };
   |         +++

warning[E0594]: cannot assign to `s.x`, as `s` is not declared as mutable
  --> $DIR/issue-35937.rs:17:5
   |
LL |     s.x += 1;
   |     ^^^^^^^^ cannot assign
   |
help: consider changing this to be mutable
   |
LL |     let mut s = S { x: 42 };
   |         +++

warning[E0594]: cannot assign to `s.x`, as `s` is not declared as mutable
  --> $DIR/issue-35937.rs:21:5
   |
LL |     s.x += 1;
   |     ^^^^^^^^ cannot assign
   |
help: consider changing this to be mutable
   |
LL | fn bar(mut s: S) {
   |        +++

warning: 3 warnings emitted

Some errors have detailed explanations: E0594, E0596.
For more information about an error, try `rustc --explain E0594`.
